clients/vmClient/vmClient.go

```diff
@@ -856,4 +857,23 @@ next:
 	return nil
 }
 
+func verifyInstanceSizeAvailableInLocation(location *locationClient.Location, vmSize string) error {
```

Comment:
Verify-> is. And this should return (bool, err) and the caller of the method should craft the error. This will allow this method to be reused in different places.